net/dnscache/dnscache.go

```diff
@@ -31,6 +31,11 @@ func preferGoResolver() bool {
 		return false
 	}
 
+	// No local resolver on Android either.
```

Comment:
Well, I added a system-wide DNS caching proxy to Android in 2011: https://github.com/aosp-mirror/platform_bionic/commit/0ec85334ff375a2b7823f322e083d4a0ea24c289 ... so each app doesn't need to do its own DNS + caching. (It just doesn't speak the normal DNS protocol that Go can speak)

I assume it or something like it is still there. But if we just do libc calls to `getaddrinfo` etc, Android's libc should do the right thing and cache the results.

But it's not like our iOS comment above is totally accurate, either, so.... *shrug*

You can submit, or fix both comments and submit. Your call.

Reply:
What I got before this change was a bunch of "connection refused" on UDP port 53:

```04-27 18:18:23.772 29228 29262 I gio     : 2020/04/27 18:18:23 derphttp.Client.Recv: connecting to https://derp4.tailscale.com/derp
04-27 18:18:23.780 29228 29262 I gio     : 2020/04/27 18:18:23 magicsock: [0xc0018a7680] derp.Recv(derp-4): derphttp.Client.Recv connect to https://derp4.tailscale.com/derp: lookup derp4.tailscale.com on [::1]:53: read udp [::1]:49923->[::1]:53: read: connection refused```

so I assumed that there wasn't a local resolver. How about

"No accessible local resolver on Android."?

Reply:
I changed it to "The local resolver is not available on Android."